activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -642,6 +644,14 @@ def first_nth_or_last(type, *args)
             set_inverse_instance record if record.is_a? ActiveRecord::Base
           end
         end
+
+        def associate_all_records(target)
+          if owner.new_record? && target.all? {|record| record.new_record? }
```

Comment:
I think it's inconsistent to treat this:

```ruby
Car.new(seats: [Seat.new, Seat.find(1)])
```

different from this:

```ruby
Car.new(seats: [Seat.new])
```

Regarding assignment of the car's id to the *new* seat in both cases.

If we assign the parent car's id to all new seats in the second case, IMO we should also extend it to the first case. 

But probably not a huge deal...